datareservoirio/authenticate.py

```diff
@@ -89,6 +91,19 @@ def __init__(self, client, auth_force=False, **kwargs):
             **kwargs,
         )
 
+        # Attention: Be careful when extending the list of retry_status!
+        retry_status = frozenset([413, 429, 502, 503, 504])
+        allowed_methods = frozenset(["GET", "POST", "PUT", "PATCH", "DELETE"])
+
+        persist = Retry(
+            total=3,
+            backoff_factor=0.5,
```

Comment:
Not to self: Check this is doubling the timeout, if so 10 second timeout to Azure can work

Reply:
It seems (Need to find the docs) that backoff_factor 0.5 will never allow timeout to increase, that it is calculated like backoff_factor*2 to allow doubling, where 0.5 will let it always be constant, therefore never allowing a call to Azure to take longer than 10 seconds, which could be too short in some cases? If there are large file uploads or downloads, or if something is paused as we have seen before with multi-threading, it would be good to allow the timeout to increase.

Reply:
I would rather decrease the timeout to DRIO to 30 seconds as default and allow a backoff_factor as 1 to increase the timeouts. Also, consider setting DEFAULT_BACKOFF_MAX from https://urllib3.readthedocs.io/en/stable/reference/urllib3.util.html to 120 seconds, more than two minutes for a call seems unlikely to succeed... But with backoff doubling and an initial value of 120 seconds, we could reach very long timeouts.

Reply:
backoff is defined as: `{backoff factor} * (2 ** ({number of previous retries}))`, which means that backoff-factor of 0.5 yields:  `(2 ** ({number of previous retries} - 1))`. That is, 1st retry waits .5 sec, second retry waits 1 sec, and then 2 secs, then 4 secs etc.

However, this may be too agressive? We have used these timout values for many years. We accidently removed them from verison 0.12.6 -> 0.12.7 or something

Reply:
DEFAULT_BACKOFF_MAX sets the absolute max backoff time regardless of any settings the user may apply.